lib/pure/collections/tables.nim

```diff
@@ -2230,6 +2225,10 @@ type
     data: seq[tuple[key: A, val: int]]
     counter: int
     countDeleted: int
+      # using this and creating tombstones as in `Table` would make `remove`
+      # amortized O(1) instead of O(n); this requires updating remove,
+      # and changing `val != 0` to what's used in Table. Or simply make CountTable
+      # a thin wrapper around Table (which would also enable `dec`, etc)
```

Comment:
> Or simply make CountTable a thin wrapper around Table (which would also enable `dec`, etc)

I don't understand this comment. Is this a `TODO`?

Reply:
I've reworded a bit this comment to clarify. I'm suggesting to possible venues to improve CountTable:
* using `countDeleted` and tombstones, and adapting the algorithms used in `Table`
* or (best option IMO, to be discussed in separate PR) using something analog to https://github.com/nim-lang/RFCs/issues/200 but for CountTable, where we offer same API as current `CountTable` but implemented as a thin wrapper around Table.

Both approaches would enable these:
* all operations would become at least as efficient as current CountTable operations
* some CountTable operations would turn from O(n) to amortized O(1)
* current API restrictions would be lifted (eg we'd allow `inc` with negative values (or equivalently, `dec`), and we'd allow counts to be 0)

You could call it a TODO, or you could call it something to help the next guy reading this code and wondering about why some operations are currently O(n). I could do that PR, after this PR is merged.